libsolidity/interface/CompilerStack.cpp

```diff
@@ -1596,10 +1599,12 @@ bytes CompilerStack::createCBORMetadata(Contract const& _contract) const
 
 	if (experimentalMode || m_viaIR)
 		encoder.pushBool("experimental", true);
-	if (m_release)
+	if (m_versionType == VersionType::Release)
 		encoder.pushBytes("solc", VersionCompactBytes);
-	else
+	else if (m_versionType == VersionType::Prerelease)
 		encoder.pushString("solc", VersionStringStrict);
+	else
+		solAssert(m_versionType == VersionType::Empty, "Invalid version type");
```

Comment:
There's an `if` at the beginning of the function so the `else` part will never be reached.
```suggestion
	else
	{
		solAssert(m_versionType == VersionType::Prerelease, "Invalid version type");
		encoder.pushString("solc", VersionStringStrict);
	}
```